src/main/java/org/threadly/litesockets/UDPClient.java

```diff
@@ -180,6 +176,31 @@ public boolean setSocketOption(final SocketOption so, final int value) {
   public ClientOptions clientOptions() {
     return uso;
   }
+  
+  @Override
+  protected void addReadBuffer(final ByteBuffer bb) {
+    addReadStats(bb.remaining());
+    synchronized(readerLock) {
+      readBuffers.add(bb);
+      callReader();
```

Comment:
As written currently it's going to go through each listener and execute on the pool for each listener while holding this lock.  I don't think there will be many listeners, but I just was not sure why you would want to hold this lock while you do that?  Unless there is a reason to lock it, I say move it out.

FWIW, there is another ListenerHelper type which will go through the listener loop on another thread, but it's for specialized cases, and I don't think it would be a good fit here.